.github/workflows/build-test-measure.yml

```diff
@@ -258,6 +315,8 @@ jobs:
 
   e2e-test-js:
     name: 'E2E test: JS'
+    needs: pre-run
+    if: needs.pre-run.outputs.changed-file-count > 0
```

Comment:
Not sure if we'd want to make this check for modified files too strict, as this touches multiple sections of the plugin.